PayPalNativeCheckout/src/main/java/com/braintreepayments/api/PayPalNativeCheckoutClient.java

```diff
@@ -1,14 +1,467 @@
+                        if (payPalAccountNonce != null) {
+                            listener.onPayPalSuccess(payPalAccountNonce);
+                        } else {
+                            listener.onPayPalFailure(new Exception("PaypalAccountNonce is null"));
+                        }
+                    });
+
+                },
+                () -> listener.onPayPalFailure(new Exception("User has canceled")),
+                errorInfo -> listener.onPayPalFailure(new Exception(errorInfo.getError().getMessage()))
+        );
+    }
+
+    private PayPalAccount setupAccount(
+        final Configuration configuration,
+        final PayPalRequest payPalRequest,
+        final PayPalResponse payPalResponse
+    ) {
+        PayPalAccount payPalAccount = new PayPalAccount();
+
+        String merchantAccountId = payPalRequest.getMerchantAccountId();
+        String paymentType = payPalRequest instanceof PayPalVaultRequest ? "billing-agreement" : "single-payment";
+        payPalAccount.setClientMetadataId(configuration.getPayPalClientId());
+        payPalAccount.setIntent(payPalResponse.getIntent());
+        payPalAccount.setSource("paypal-browser");
+        payPalAccount.setPaymentType(paymentType);
+
+        if (merchantAccountId != null) {
+            payPalAccount.setMerchantAccountId(merchantAccountId);
+        }
+
+        return payPalAccount;
+    }
+
+    private static String getAnalyticsEventPrefix(PayPalRequest request) {
+        return request instanceof PayPalVaultRequest ? "paypal.billing-agreement" : "paypal.single-payment";
+    }
+
+    void onBrowserSwitchResult(FragmentActivity activity) {
+        this.pendingBrowserSwitchResult = braintreeClient.deliverBrowserSwitchResult(activity);
+
+        if (pendingBrowserSwitchResult != null && listener != null) {
+            deliverBrowserSwitchResultToListener(pendingBrowserSwitchResult);
+        }
+    }
+
+    private void deliverBrowserSwitchResultToListener(final BrowserSwitchResult browserSwitchResult) {
+        onBrowserSwitchResult(browserSwitchResult, (payPalAccountNonce, error) -> {
+            if (payPalAccountNonce != null) {
+                listener.onPayPalSuccess(payPalAccountNonce);
+            } else if (error != null) {
+                listener.onPayPalFailure(error);
+            }
+        });
+        this.pendingBrowserSwitchResult = null;
+    }
+
+    BrowserSwitchResult getBrowserSwitchResult(FragmentActivity activity) {
+        return braintreeClient.getBrowserSwitchResult(activity);
+    }
+
+    /**
+     * Deprecated. Use {@link PayPalListener} to handle results.
+     *
+     * @param browserSwitchResult a {@link BrowserSwitchResult} with a {@link BrowserSwitchStatus}
+     * @param callback            {@link PayPalBrowserSwitchResultCallback}
+     */
+    @Deprecated
+    public void onBrowserSwitchResult(@NonNull BrowserSwitchResult browserSwitchResult, @NonNull final PayPalBrowserSwitchResultCallback callback) {
```

Comment:
Do we still need this code?

Reply:
No we don't, removed all the browser switch code